clients/unshaded/src/main/java/tachyon/client/RemoteBlockOutStream.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Closer;
+import com.google.common.primitives.Ints;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.util.CommonUtils;
+
+/**
+ * <code>RemoteBlockOutStream</code> implementation of TachyonFile. This class is not client facing.
+ */
+public class RemoteBlockOutStream extends BlockOutStream {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private final int mBlockIndex;
+  private final long mBlockId;
+  private final long mBlockCapacityBytes;
+  private final Closer mCloser;
+  private final RemoteBlockWriter mRemoteWriter;
+
+  // The local write buffer to accumulate client writes.
+  private final ByteBuffer mBuffer;
+  // The size of the write buffer in bytes.
+  private final long mBufferBytes;
+  // Total number of bytes written to block, but not necessarily flushed to the remote server.
+  private long mWrittenBytes = 0;
+  // True if stream is open.
+  private boolean mOpen = false;
```

Comment:
Can we change this to `mClosed` and reverse the logic to be consistent with the rest of the codebase?